source/set-readonly.d.ts

```diff
@@ -0,0 +1,33 @@
+import type {Except} from './except';
+import type {Simplify} from './simplify';
+
+/**
+Create a type that makes the given keys readonly. The remaining keys are kept as is.
+
+Use-case: You want to define a single model where the only thing that changes is whether or not some of the keys are readonly.
+
+@example
+```
+import type {SetReadonly} from 'type-fest';
+
+type Foo = {
+	a: number;
+	readonly b: string;
+	c: boolean;
+}
+
+type SomeReadonly = SetReadonly<Foo, 'b' | 'c'>;
+// type SomeReadonly = {
+// 	a: number;
+// 	readonly b: string; // Was already readonly and still is.
+// 	readonly c: boolean; // Is now readonly.
+// }
+```
+
+@category Object
+*/
+export type SetReadonly<BaseType, Keys extends keyof BaseType> =
+	Simplify<
+	Except<BaseType, Keys> &
+	Readonly<Pick<BaseType, Keys>>
+	>;
```

Comment:
Would be great if you could implement the same fix as this: https://github.com/sindresorhus/type-fest/commit/6a82900b45c7c2bcf3f5964dd07bc5e8a3999dac

Reply:
done